Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -96,10 +111,17 @@ public struct MqttRequestResponse: Sendable {
 /// A response path is a pair of values - MQTT topic and a JSON path - that describe where a response to
 /// an MQTT-based request may arrive.  For a given request type, there may be multiple response paths and each
 /// one is associated with a separate JSON schema for the response body.
-public class ResponsePath: CStruct, @unchecked Sendable {
+final public class ResponsePath: CStruct, @unchecked Sendable {
+  /// MQTT topic that a response may arrive on.
   let topic: String
+  /// JSON path for finding correlation tokens within payloads that arrive on this path's topic.
```

Comment:
This feels misleading because it's labeled here and elsewhere as "JsonPath" but it's actually just the unique string used to correlate. Unsure what we should do from a naming the member perspective but I don't think we should say it's a "JSON path" in the description we provide.

Reply:
Calling it a "path" still feels accurate to me. Currently, the correlation token is just a top-level string, so the "path" is simply "clientToken" like in this example:
```
Response{
  "clientToken": "<token>"
}
```
But our current design leaves room for future responses where the token could be nested. For example, in the following case, the path might be something like "level1.level2.clientToken".
```
Response{
  "level1": {
    "level2" : {
        "clientToken": "<token>"
    }
  }
}
```